blocks/serializer/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import * as query from 'hpq';
+
+/**
+ * Takes a block list and returns the serialized post content
+ *
+ * @param  {Array}  blocks Block list
+ * @return {String}        The post content
+ */
+export default function serialize( blocks ) {
+	return blocks.reduce( ( memo, block ) => {
+		const blockType = block.blockType;
+		const blockSettings = wp.blocks.getBlockSettings( blockType );
+		const rawContent = wp.element.renderToString(
+			blockSettings.save( block.attributes )
```

Comment:
The initial direction was decided primarily from the perspective of how it's used in implementing a block, since the duality between `edit` and `save`, both with a consistent return type, seemed the most direct way to communicate the intent for the implementer to create an incoming (editor) and outgoing (to save) description of the UI.

Alternative names to consider:

- `prepareForEdit`, `prepareForSave` (not entirely accurate since it's not preparing, it's returning the element describing the UI)
- `renderEdit`, `renderSave`

Honestly it doesn't really bother me that it looks awkward in the one internal snippet of code where we're actually invoking it. One change I'd like to make which could improve this is to treat `save` and `edit` as components, not mere functions, like started in #335:

https://github.com/WordPress/gutenberg/pull/335/files#diff-1b2cbb4a0a02cacc1be65d67ece68c21

>return the string directly in the function.

If we upgrade to `react@next` (16 w/ Fiber), strings are valid return values of a component's (function's) render, so we could support string as an option without making any distinction.